### DISCUSSION TITLE
Hide lines at the start of a code block?

### DISCUSSION THREAD

**Discussion by conery** [mattcoleanderson/obsidian-conceal-plugin#32] [Q&A] [answered]
Hi -- 

I use Obsidian exclusively in live preview mode.  One quirk that I'd like to get around is that in preview mode the first and last line of a code block are always displayed, and this adds a lot of ugly blank space when the block is not being edited.  

I thought I'd try your plugin.  It should be possible to write a regexp that looks for three backticks at the front of a line.  In Python, it's `r'^```'`.

I just submitted an issue -- if I try to type that regexp into the plugin Obsidian freezes as soon as I type the caret.

If I do get that going, do you think it will do what I want and hide the first and last lines of a code block?

**Comment by mattcoleanderson** [comment, marked as answer]
@conery I have written an explanation of the issue in #31 

As far as hiding the line, its not possible with this plugin.

This plugin only conceals textual information that is visible in preview mode. The generate HTML is parsed for the regex and the pattern is removed from it. Codeblocks in Obsidian work by using a CodeMirror plugin that replaces the 3 back ticks in Source mode with html representing the visual code block you se see in Preview mode. This means that the 3 backticks are actually absent in the html for Preview mode when the the Conceal plugin is parsing it.

You'd have a better time writing a custom css file for styling code blocks to your preference. You could find the element that creates the blank line and change its height, for example.

**Comment by conery** [comment]
Thanks for the explanation, Matt!

John
